core/src/main/java/org/elasticsearch/indices/cluster/IndicesClusterStateService.java

```diff
@@ -217,11 +248,15 @@ private void applyDeletedIndices(final ClusterChangedEvent event) {
             if (logger.isDebugEnabled()) {
                 logger.debug("[{}] cleaning index, no longer part of the metadata", index);
             }
-            final IndexService idxService = indicesService.indexService(index);
+            AllocatedIndex<? extends Shard> indexService = indicesService.indexService(index);
             final IndexSettings indexSettings;
-            if (idxService != null) {
-                indexSettings = idxService.getIndexSettings();
-                deleteIndex(index, "index no longer part of the metadata");
+            if (indexService != null) {
+                indexSettings = indexService.getIndexSettings();
```

Comment:
in a different change - since were talking about it - if we inline pending delete processing with Indices service, where I grow more to think it naturally belong, this indexSetting juggling will go away